components/FormInput/SharedFormInput.js

```diff
@@ -21,6 +21,26 @@ export default class SharedFormInput extends Component {
       focus: false,
       id: uniqueId('forminput'),
     };
+
+    this.listener = null;
+  }
+
+  componentDidMount() {
+    this.listener = setInterval(() => {
```

Comment:
This is fine. All I would say is it might be slightly ... 'sexier' if:
- you pulled the anonymous function out and declared it on the component
- you declared the `const e` at the top of the scope. Good habit to get into because it avoids issues caused by the temporal dead zone.

So I'd go for something like:
```
componentDidMount() {
    this.listener = setInterval(this.watchInputs, 20);
}

watchInputs = () => {
    const input = ReactDOM.findDOMNode(this.refs.focusable);
    const e = document.createEvent('HTMLEvents'); // Move to top of scope

    // ... rest of function
}
```